### PR TITLE
GNUmakefile: add spellfix target, use it.

### PR DESCRIPTION
**Pull request by dkegel-fastly**
Also:
- point to an updated version of the misspell package.
- give internal/tools its own go.mod temporarily because new misspell needs a newer go than toplevel go.mod specifies
- make lint target depend on tools target and use the revive it installs (assuming go install causes it to be found via PATH)
- fix what newer misspell and lint find
- add spell to CI

### CODE REVIEW

File: compiler/compiler.go [hunk 1]
```diff
@@ -242,7 +242,7 @@ func NewTargetMachine(config *Config) (llvm.TargetMachine, error) {
 }
 
 // Sizes returns a types.Sizes appropriate for the given target machine. It
-// includes the correct int size and aligment as is necessary for the Go
+// includes the correct int size and allignment as is necessary for the Go
```
Comment (dgryski) [inline compiler/compiler.go]: `alignment`

File: src/internal/wasi/clocks/v0.2.0/monotonic-clock/monotonic-clock.wit.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func wasmimport_Resolution() (result0 uint64)
 // SubscribeInstant represents the imported function "subscribe-instant".
 //
 // Create a `pollable` which will resolve once the specified instant
-// occured.
+// occurred.
```
Comment (dgryski) [inline src/internal/wasi/clocks/v0.2.0/monotonic-clock/monotonic-clock.wit.go]: Sadly this is a misspelling upstream, *and* this is a generated file.  We should probably ignore `src/internal/wasi` for now.

File: src/internal/wasi/io/v0.2.0/streams/streams.wit.go [hunk 3]
```diff
@@ -413,7 +413,7 @@ func wasmimport_OutputStreamFlush(self0 uint32, result *cm.Result[StreamError, s
 //
 // Read from one stream and write to another.
 //
-// The behavior of splice is equivelant to:
+// The behavior of splice is equivalent to:
```
Comment (dgryski) [inline src/internal/wasi/io/v0.2.0/streams/streams.wit.go]: I might submit spelling fixes for the original `.wit` files upstream.

**Review by dgryski — CHANGES_REQUESTED** [review]
I'm a fan of automation.  Even better if the tooling is reliable enough to run as part of CI.

File: compiler/compiler.go (hunk 1 above)
Comment (dkegel-fastly) [reply compiler/compiler.go]: unbroke

File: src/internal/wasi/clocks/v0.2.0/monotonic-clock/monotonic-clock.wit.go (hunk 2 above)
Comment (dkegel-fastly) [reply src/internal/wasi/clocks/v0.2.0/monotonic-clock/monotonic-clock.wit.go]: ignored

**Comment by dkegel-fastly** [comment]
Thanks for the review!  Ready for another look.

**Review by dgryski — APPROVED** [review]
LGTM

File: go.mod [hunk 4]
```diff
@@ -1,14 +1,16 @@
 module github.com/tinygo-org/tinygo
 
-go 1.19
+go 1.21
```
Comment (b0ch3nski) [inline go.mod]: I think I saw somewhere on tinygo.org docs mentioning a requirement of Go 1.20 - would be good to confirm that and also bump the version in docs

**Review by archie2x — APPROVED** [review]
LGTM

File: go.mod
Comment (leongross) [inline go.mod:1]: Ci [fails](https://github.com/tinygo-org/tinygo/actions/runs/10288011952/job/28472435670?pr=4387#step:19:39) with
```
go: updates to go.mod needed; to update it:
	go mod tidy
```
So update go modules and/or rebase against dev?

**Review by leongross — CHANGES_REQUESTED** [review]

File: go.mod
Comment (dkegel-fastly) [reply go.mod:1]: I suspect separate go.mod to specify a newer go for now, to be removed when top level go.mod catches up

File: go.mod (hunk 4 above)
Comment (dkegel-fastly) [reply go.mod]: Not yet!  I think we still work with go 1.19.  So the proper workaround is probably a separate go.mod just for the spell checker for the moment.

**Review by dgryski — APPROVED** [review]
LGTM

**Pull request merged**